nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RetryFlowFile.java

```diff
@@ -0,0 +1,225 @@
+        "If the FlowFile's attribute value exceeds the configured maximum, the FlowFile will be passed to a " +
+        "'retries_exceeded' relationship. WARNING: If the incoming FlowFile has a non-numeric value in the " +
+        "configured 'Retry Attribute' attribute, it will be reset to '1'. You may choose to fail the FlowFile " +
+        "instead of performing the reset. Additional dynamic properties can be defined for any attributes you " +
+        "wish to add to the FlowFiles transferred to 'retries_exceeded'. These attributes support attribute " +
+        "expression language.")
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@SupportsBatching
+@SideEffectFree
+@DefaultSettings(penaltyDuration = "2 min")
+@ReadsAttribute(attribute = "Retry Attribute",
+        description = "Will read the attribute or attribute expression language result as defined in 'Retry Attribute'")
+@WritesAttribute(attribute = "Retry Attribute",
+        description = "User defined retry attribute is updated with the current retry count")
+@DynamicProperty(name = "Exceeded FlowFile Attribute Key",
+        value = "The value of the attribute added to the FlowFile",
+        description = "One or more dynamic properties can be used to add attributes to FlowFiles passed to " +
+                "the 'retries_exceeded' relationship",
+        expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+public class RetryFlowFile extends AbstractProcessor {
+    private List<PropertyDescriptor> properties;
+    private Set<Relationship> relationships;
+    private String retryAttribute;
+    private Integer maximumRetries;
+    private Boolean penalizeRetried;
+    private Boolean failOnOverwrite;
+
+    public static final PropertyDescriptor RETRY_ATTRIBUTE = new PropertyDescriptor.Builder()
+            .name("Retry Attribute")
+            .description("The name of the attribute that contains the current retry count for the FlowFile. " +
+                    "WARNING: If the name matches an attribute already on the FlowFile that does not contain a " +
+                    "numerical value, the processor will either overwrite that attribute with '1' or fail " +
+                    "based on configuration.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_EL_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .defaultValue("flowfile.retries")
+            .build();
+    public static final PropertyDescriptor MAXIMUM_RETRIES = new PropertyDescriptor.Builder()
+            .name("Maximum Retries")
+            .description("The maximum number of times a FlowFile can be retried before being " +
+                    "passed to the 'retries_exceeded' relationship")
+            .required(true)
+            .addValidator(StandardValidators.createLongValidator(1, Integer.MAX_VALUE, true))
+            .defaultValue("3")
+            .build();
```

Comment:
Couldn't figure out how to do this cleanly since it would require checking that I have a number on every onTrigger execution.

Reply:
Yeah, you'd need to:
- Change instance variable `maximumRetries` to a local variable inside `onTrigger`
- Evaluate EL at `onTrigger` using the current FlowFile

I think having different max retry per FlowFile may be a valid use-case. Especially where a common retry route is used from different places.

Reply:
Indeed that might be valuable. Is there a quick and dirty way to parse number using a utility class within NiFi that I'm not aware of? I would hate to wrap it in a NumberFormatException for every trigger...

Reply:
Nevermind, dumb question. I'll implement the change and repush

Reply:
Alright I have implemented the changes. Note that because asInteger doesn't handle not integer expressions I had to wrap it in a NumberFormatException. :(